### PR TITLE
Add generation preconditions to UploadObject

### PR DESCRIPTION
**Pull request by jskeet**

### CODE REVIEW

File: GoogleApiWrappers/src/Google.Apis.Storage.v1.ClientWrapper/UploadObjectOptions.cs [hunk 1]
```diff
@@ -18,6 +19,30 @@ public class UploadObjectOptions
...
+        public long? IfGenerationMatch;
+
+        /// <summary>
+        /// Precondition for upload: if the object's generation matches the
+        /// given generation, it is not uploaded.
+        /// </summary>
+        public long? IfGenerationNotMatch;
+
+        /// <summary>
+        /// Precondition for upload: if the object's meta-generation does not match the
```
Comment (ivannaranjo) [inline GoogleApiWrappers/src/Google.Apis.Storage.v1.ClientWrapper/UploadObjectOptions.cs]: I have to say, the double negative makes it a bit confusing.

**Comment by ivannaranjo** [comment]
Looks good to me, only the double negative in the command seems a bit confusing, but that's about it. :shipit:

File: GoogleApiWrappers/src/Google.Apis.Storage.v1.ClientWrapper/UploadObjectOptions.cs (hunk 1 above)
Comment (jskeet) [inline GoogleApiWrappers/src/Google.Apis.Storage.v1.ClientWrapper/UploadObjectOptions.cs]: Have changed all the comments - let me know what you think. I'll amend the download ones in the same way if you like these.

**Comment by ivannaranjo** [comment]
Love it :)

**Pull request merged**